package.go

```diff
@@ -4,17 +4,17 @@ type Package struct {
 	// unique ID of this package. this will be created as discovered by the library
 	// and used for persistence and hash map indexes
 	ID int `json:"id"`
-	// the name of the distribution
+	// the name of the package
 	Name string `json:"name"`
-	// the version of the distribution
+	// the version of the package
 	Version string `json:"version"`
 	// type of package. currently expectations are binary or source
-	Kind string `json:"kind"`
+	Kind string `json:"kind,omitempty"`
 	// if type is a binary package a source package maybe present which built this binary package.
 	// must be a pointer to support recursive type:
-	Source *Package `json:"source"`
+	Source *Package `json:"source,omitempty"`
```

Comment:
omit empty can break downstream clients no? I usually avoid the need to explain "this field might be there" to ppl utilizing structs via json.

Reply:
this can already be there or not, because it's a pointer. this just makes the resulting json smaller.

Reply:
@hdonnay the pointer is used to support a recursive datastructure - not so much to support a nil value. 

I can see the point, but won't this make clients more brittle ? And our openapi docs less reliable? I guess right now we return an empty value for it.